lsp-diagnostics.el

```diff
@@ -0,0 +1,314 @@
+;;; lsp-diagnostics.el --- LSP diagnostics integration -*- lexical-binding: t; -*-
+;;
+;; Copyright (C) 2020 emacs-lsp maintainers
+;;
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+;;
+;;; Commentary:
+;;
+;;  LSP diagnostics integration
+;;
+;;; Code:
+
+(require 'lsp-mode)
+
+(define-obsolete-variable-alias 'lsp-prefer-flymake
+  'lsp-diagnostics-provider  "lsp-mode 6.2")
+
+(define-obsolete-variable-alias 'lsp-diagnostic-package
+  'lsp-diagnostics-provider  "lsp-mode 7.0.1")
```

Comment:
This one I am not 100% it will work:

e. g. 
```
(use-package lsp-mode
   :config (setq lsp-diagnostics-package 'foobar)
```

The setq will be executed after lsp-mode.el is loaded but at this point lsp-diagnostics.el won't be loaded and the alias won't be loaded. I think the result will be that the `provider` wont be set. I see several options to avoid that:

1. Move the defalias temporary(until we drop the alias) in lsp-mode.el
2. Keep the original name and rename after several months/releases.
?

Reply:
yeah, but I think the problem is bigger than that, even moving to `lsp-mode.el`, `lsp-diagnostics.el` won't be loaded when this `(setq)` run, right?

Reply:
> yeah, but I think the problem is bigger than that, even moving to `lsp-mode.el`, `lsp-diagnostics.el` won't be loaded when this `(setq)` run, right?

Yes, but setting symbol before the defining package is loaded is fine. The issue when you set an alias of a symbol that is not loaded.

Reply:
Another options is to use autoload so this will be loaded even before package is loaded.

Reply:
Some thing like
``` elisp
;;;###autoload
(define-obsolete-variable-alias 'a 'b "now")
```

Reply:
This will most likely work.

Reply:
Good idea, I'll do that